### ISSUE TITLE
`pipx list` should have an option to list packages only

### ISSUE THREAD

**Issue by njthomas** [pypa/pipx#792]
**How would this feature be useful?**
Currently, `pipx list` outputs a long/verbose format by default:
```
$ pipx list
venvs are in /usr/home/njt/code/3p/pipx/venvs
apps are exposed on your $PATH at /usr/home/njt/code/3p/pipx/bin
   package pipx 0.17.0, installed using Python 3.8.12
    - pipx
   package virtualenv 20.12.1, installed using Python 3.8.12
    - virtualenv
```

This is hard to parse visually (and programatically, but one could argue that `--json` exists for that purpose, but more on that below).

There is a `--verbose` option available, but it adds information to an already verbose output format.

**Describe the solution you'd like**
I would like pipx to support a `--short` option to list the installed packages only:
```
$ pipx list --short
pipx
virtualenv
```

**Describe alternatives you've considered**

Ideally, `pipx list` by itself would have given the short listing by default, and `--verbose` would have given the extra information, but since this is already set, and users are used to the listing as it is, adding `--short` to do this would suffice.

**NB1:** Since pipx has not yet reached v1.0, I don't think it would be too big a deal to have `list` print the short output by default, and move the current default to `--verbose`, but I will leave that up to the maintainers to decide if they want to do that. I don't think it would break anything (users should be using `--json` to parse output anyway, and according to [Semantic Versioning rules](https://semver.org/#spec-item-4), major version zero allows for breaking changes.

**NB2:** I am aware that using `--json` and parsing the output will give the same effect (ie. `pipx list --json|jq -r '.venvs|keys[]'`), however, I submit to you that querying for the installed packages only **is a very common operation** and users should not have to resort to parsing JSON output to get this information.

**NB3:** I am okay with calling the option `--packages-only` or anything else, I just used `--short` as an example.

**Comment by gaborbernat** [comment]
> **NB1:** Since pipx has not yet reached v1.0,

You missed that train by like 8 hours, https://twitter.com/gjbernat/status/1478297177604698115

> I would like pipx to support a `--short` option to list the installed packages only:

I'd be open to this, feel free to put in a PR with tests for this. I'd perhaps suggest to use PEP-508 formatting to display the version number in there, but that's just an option.

**Comment by vrslev** [comment]
Maybe use flag “-q” instead?

**Comment by cs01** [comment]
Closed by https://github.com/pypa/pipx/pull/804
